server/controller/apis/topicController.js

```diff
@@ -5,10 +5,72 @@
  * see included LICENSE or https://opensource.org/licenses/BSD-3-Clause 
  */
 
+// dependencies
+const fs = require( 'fs' );
+let path = require( 'path' );
+
 // import services
-const topicService = require( '../service/topicService' );
+const topicService = require( '../../service/topicService' );
+
+// import util Models
+const ApiMessage = require( '../../model/util/ApiMessage' );
 
 // import models
-const CompletedResource = require( '../model/completedResource' );
+const Topic = require('../../model/topic' );
+
+// set up file paths for user profile images
+const UPLOAD_PATH_BASE = path.resolve( __dirname, '..', '../../client' );
+const FRONT_END = process.env.FRONT_END_NAME;
+let TOPIC_PATH = process.env.TOPIC_IMAGE_PATH;
+
+// topic file path
+const topicUploadPath = UPLOAD_PATH_BASE + "/" + FRONT_END + TOPIC_PATH;
+
+// set the max image size for avatars and resource, topic and goal icons
+const maxSize = process.env.IMAGE_UPLOAD_MAX_SIZE;
+const maxSizeText = process.env.IMAGE_UPLOAD_MAX_SIZE_FRIENDLY_TEXT;
+
+exports.getAllVisibleTopics = async ( req, res ) => {
+
+    // get the auth user id from either the basic auth header or the session
+    let authUserId;
+    if( req.user ) {
+        authUserId = req.user.id;
+    }
+    else if( req.session.authUser ) {
+        authUserId = req.session.authUser.id;
+    }
```

Comment:
Per Professor Gormanly's suggestion in #176, we should include an API message to return 401 on authorization failure.

```suggestion
    } else {
        const message = ApiMessage.createApiMessage( 401, "Not Authorized", "Not able to associate authorized user with the record" );
        res.set( "x-agora-message-title", "Not Authorized" );
        res.set( "x-agora-message-detail", "Not able to associate authorized user with the record" );
        res.status( 401 ).json( message );
    }
```

Reply:
I reached out to our PM to create a ticket regarding this across all the APIs. We can focus on this in Milestone 2

Reply:
Thanks @Mike-Volchko. Additionally, I want to add 204 No Content status code validation to resources upon to getAllVisibleResources. Although it can be done quickly, to avoid conflicts and induce clarity, I believe this should also be included with this ticket that is being created by PM.

Reply:
If it is confirmed that this will be handled separately then this PR can continue to completion